crates/node-api/src/engine/error.rs

```diff
@@ -4,30 +4,46 @@ use thiserror::Error;
 /// Thrown when the payload or attributes are known to be invalid before processing.
 #[derive(Error, Debug)]
 pub enum AttributesValidationError {
-    /// Thrown if `PayloadAttributes` provided in engine_forkchoiceUpdated before V3 contains a
-    /// parent beacon block root
+    /// Thrown when the underlying validation error occured while validating an `ExecutionPayload`.
+    #[error("Payload validation error: {0}")]
+    Payload(PayloadOrAttributesValidationError),
+
+    /// Thrown when the underlying validation error occured while validating a `PayloadAttributes`.
+    #[error("Payload attributes validation error: {0}")]
+    PayloadAttributes(PayloadOrAttributesValidationError),
+
+    /// Thrown if `PayloadAttributes` or `ExecutionPayload` were provided with a timestamp, but the
+    /// version of the engine method called is meant for a fork that occurs after the provided
+    /// timestamp.
+    #[error("Unsupported fork")]
+    UnsupportedFork,
+    /// Another type of error that is not covered by the above variants.
+    #[error("Invalid params: {0}")]
+    InvalidParams(#[from] Box<dyn std::error::Error + Send + Sync>),
+}
+
+/// Thrown when validating a payload or attributes fails.
+#[derive(Error, Debug)]
+pub enum PayloadOrAttributesValidationError {
```

Comment:
can we make this a bit more clear that his is for both: OR
```suggestion
/// Thrown when validating an execution payload OR payload attributes fails.
#[derive(Error, Debug)]
pub enum PayloadOrAttributesValidationError {
```